app/src/main/java/io/github/trojan_gfw/igniter/ProxyService.java

```diff
@@ -185,38 +151,13 @@ private void shutdown() {
             Log.e("Clash", "clash stopped");
         }
         Tun2socks.stop();
```

Comment:
Is `Tun2socks.stop();` a block or non-block call? Is it possible to cause the UI reenabled while the `stopping` is still  in progress?

Reply:
Moreover, the tun device could be closed by system(tun off VPN in system setting). Is this situation considered?

Reply:
A blocking call. The button is disabled while the stopping is in progress.

Reply:
Oh, the correct words are "sync call" and "async call", lol